File: server/src/domain/entities/FeatureFlag.ts
export class FeatureFlag {
  constructor(
    public readonly id: number,
    public readonly featureKey: string,
    public readonly featureName: string,
    public readonly description: string,
    public readonly enabled: boolean,
    public readonly updatedAt: Date
  ) {
    this.validate();
  }

  private validate(): void {
    if (!this.featureKey || this.featureKey.trim().length === 0) {
      throw new Error('Feature key is required');
    }
    
    if (!this.featureName || this.featureName.trim().length === 0) {
      throw new Error('Feature name is required');
    }
    
    if (!this.description || this.description.trim().length === 0) {
      throw new Error('Feature description is required');
    }
  }

  toggle(): FeatureFlag {
    return new FeatureFlag(
      this.id,
      this.featureKey,
      this.featureName,
      this.description,
      !this.enabled,
      new Date()
    );
  }

  enable(): FeatureFlag {
    return new FeatureFlag(
      this.id,
      this.featureKey,
      this.featureName,
      this.description,
      true,
      new Date()
    );
  }

  disable(): FeatureFlag {
    return new FeatureFlag(
      this.id,
      this.featureKey,
      this.featureName,
      this.description,
      false,
      new Date()
    );
  }

  static create(params: Omit<FeatureFlag, 'id' | 'updatedAt'> & { id?: number; updatedAt?: Date }): FeatureFlag {
    const id = params.id || Date.now();
    const updatedAt = params.updatedAt || new Date();
    
    return new FeatureFlag(
      id,
      params.featureKey,
      params.featureName,
      params.description,
      params.enabled,
      updatedAt
    );
  }
}

File: go-server/internal/models/feature_flag.go
package models

// FeatureFlag represents a feature toggle in the application.
type FeatureFlag struct {
	ID          int    `json:"id"`
	FeatureKey  string `json:"featureKey"`
	FeatureName string `json:"featureName"`
	Description string `json:"description"`
	Enabled     bool   `json:"enabled"`
	UpdatedAt   string `json:"updatedAt"`
}

// InsertFeatureFlag contains the fields required to create a new feature flag.
type InsertFeatureFlag struct {
	FeatureKey  string `json:"featureKey"`
	FeatureName string `json:"featureName"`
	Description string `json:"description"`
	Enabled     *bool  `json:"enabled"`
}
